command/kv_helpers.go

```diff
@@ -197,3 +199,67 @@ func padEqualSigns(header string, totalLen int) string {
 
 	return fmt.Sprintf("%s %s %s", strings.Repeat("=", equalSigns/2), header, strings.Repeat("=", equalSigns/2))
 }
+
+// listSecretsRecursive dfs-traverses the secrets tree rooted at the given path
+// and returns a list of the leaf paths. Note: for kv-v2, a "metadata" path is
+// expected and "metadata" paths will be returned. If includeDirectories is
+// specified, the output will include non-leaf nodes with "/" suffixes.
+func listSecretsRecursive(ctx context.Context, client *api.Client, path string, includeDirectories bool) ([]string, error) {
```

Comment:
this is potentially quite a large result, would a channel be better?

Reply:
I'm returning just a slice of strings containing paths so I don't think memory should be much of an issue here. I was thinking to have it take a `func (path string)` functor instead (visitor pattern) but it would not be as convenient 🤷

Reply:
On a second thought, I've switched this to the visitor pattern :)